### PR TITLE
Remove RestoreSources from Tools.props

### PR DESCRIPTION
**Pull request by ViktorHofer**
Related to https://github.com/dotnet/arcade/pull/3041

### CODE REVIEW

File: eng/Tools.props [hunk 1]
```diff
@@ -1,14 +1,7 @@
 <Project>
   <PropertyGroup>
-    <RestoreSources Condition="'$(DotNetBuildOffline)' != 'true'">
-      https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json;
```
Comment (safern) [inline eng/Tools.props]: Do these have to be added to the repo `NuGet.config`?

Comment (ViktorHofer) [reply eng/Tools.props]: No, the dotnet-buildtools one is auto added by arcade and the sourcelink one isn't necessary anymore.

**Comment by ViktorHofer** [comment]
https://dnceng.visualstudio.com/internal/_build/results?buildId=261058

**Comment by ViktorHofer** [comment]
@safern there were warnings in the build which I need to address. Will open a new PR.

**Pull request merged**